### DISCUSSION TITLE
Do we need to access previous state when updating state?

### DISCUSSION THREAD

**Discussion by AlexisN94** [pmndrs/zustand#1296] [Q&A]
There's already [a question](https://github.com/pmndrs/zustand/discussions/821) regarding _how_ to access the previous state when updating the state, like this:
````
const {count, setCount} = CountStore()
setCount((count) => count + 1)
````

And I know that when using React's setState, it's better to do it in this manner to avoid asynchronism issues. But is this also necessary with Zustand? Or is it internally already ensuring that the state is always updated? So that we can just do this:
````
const {count, setCount} = CountStore()
setCount(count + 1)
````

**Comment by dai-shi** [comment]
It's the same thing. `count` can be stale denending on when you invoke `setCount(count + 1)`.

---
fwiw, `setCount((count) => count + 1)` and `setCount(CountStore.getState().count + 1)` are equivalent.
